examples/scripts/rmse_estimation.py

```diff
@@ -30,31 +31,23 @@
 ]
 
 # Define the cost to optimise
-cost = pybop.RMSE()
-signal = "Voltage [V]"
+signal = "Terminal voltage [V]"
+problem = pybop.Problem(model, parameters, dataset, signal=signal, init_soc=0.98)
+cost = pybop.RootMeanSquaredError(problem)
 
 # Build the optimisation problem
-parameterisation = pybop.Optimisation(
-    cost=cost,
-    model=model,
-    optimiser=pybop.NLoptOptimize(n_param=len(parameters)),
-    parameters=parameters,
-    dataset=dataset,
-    signal=signal,
-    init_soc=0.97,
-)
+parameterisation = pybop.Optimisation(cost=cost, optimiser=pybop.NLoptOptimize)
 
 # Run the optimisation problem
-x, output, final_cost, num_evals = parameterisation.run()
-
-# get MAP estimate, starting at a random initial point in parameter space
-# parameterisation.map(x0=[p.sample() for p in parameters])
-
-# or sample from posterior
-# parameterisation.sample(1000, n_chains=4, ....)
-
-# or SOBER
-# parameterisation.sober()
-
-
-# Optimisation = pybop.optimisation(model, cost=cost, parameters=parameters, observation=observation)
+x, final_cost = parameterisation.run()
```

Comment:
so I'm guessing that `run` returns a numpy array for `x`, and `problem.evaluate` takes a numpy array and returns a numpy array. I've never really liked this api for pints, because I felt as though there should be some labelling on which parameter was which (i.e. take and return a dict like object), and for the simulated values if you've got multiple outputs then you have to remember the right shape, what dimension is time and which is outputs. Plus the cost function can introduce multiple additional parameters which the user has not specified (e.g. sigma of noise). Just a comment for discussion, numpy arrays are fine, but perhaps we want to have something a bit more structured to add metadata like names.

Reply:
Thanks for the review Martin. I agree, passing the numpy array back does lack information for the end-user. We should be able to update the parameter dictionary with the optimal value and return the entire dict. I'll add this as a future issue :)